frictionless/actions/validate.py

```diff
@@ -103,6 +105,12 @@ def validate_package(
     # Create state
     timer = helpers.Timer()
 
+    # Read resource name for single resource validation
+    resource_name = None
```

Comment:
I was wondering if we can just use a simpler logic:
- if `resource_name` is provided -> return from `resource.validate` function for this resource

Not sure if it's possible regarding options etc

Reply:
I have tried revising the code logic:

- The field_name has to be removed from options else it generates error in [line 123](https://github.com/frictionlessdata/frictionless-py/pull/1112/commits/921203d6c461e1991a15f848d17dce74a9ed4132#diff-08a6fec69d71afaa32a62bdda421723363f8235f66a588d3676dc717d639bb9dR123):


 However I could not think of any other solution for the filtering part of resources because the resources is being used in many different places:
- Validating
- Task error generation (parallel and sequential)

So I put it in a different helper function .